File: js/test/test.ExtensionsLoadUnpackedInjector.ts
import assert from "node:assert/strict";
import { existsSync } from "node:fs";
import path from "node:path";
import { test } from "vitest";

import { ExtensionsLoadUnpackedInjector } from "../src/injector/ExtensionsLoadUnpackedInjector.js";

test("ExtensionsLoadUnpackedInjector prepares the default packaged extension zip", async () => {
  const injector = new ExtensionsLoadUnpackedInjector();

  try {
    await injector.prepare();
    const unpacked_extension_path = (injector as unknown as { unpacked_extension_path?: string | null })
      .unpacked_extension_path;
    assert.equal(typeof unpacked_extension_path, "string");
    assert.match(unpacked_extension_path!, /modcdp-extension-/);
    assert.equal(existsSync(path.join(unpacked_extension_path!, "manifest.json")), true);
  } finally {
    await injector.close();
  }
});

File: python/tests/test_ExtensionsLoadUnpackedInjector.py
from __future__ import annotations

import unittest
from pathlib import Path
from typing import cast

from modcdp.injector.ExtensionsLoadUnpackedInjector import ExtensionsLoadUnpackedInjector


class ExtensionsLoadUnpackedInjectorTests(unittest.TestCase):
    def test_prepares_default_packaged_extension_zip(self) -> None:
        injector = ExtensionsLoadUnpackedInjector()
        try:
            injector.prepare()
            unpacked_extension_path = injector.unpacked_extension_path
            self.assertIsInstance(unpacked_extension_path, str)
            unpacked_extension_path = cast(str, unpacked_extension_path)
            self.assertIn("modcdp-extension-", unpacked_extension_path)
            self.assertTrue((Path(unpacked_extension_path) / "manifest.json").exists())
        finally:
            injector.close()


if __name__ == "__main__":
    unittest.main()
